### DISCUSSION TITLE
Set the order parameter in iterative_train_test_split

### DISCUSSION THREAD

**Discussion by edufonseca** [scikit-multilearn/scikit-multilearn#285] [Q&A]
Thanks for the effort on buliding this library. I have 2 questions:

**a)** The `IterativeStratification` method has a parameter `order`:

_the order of label relationship to take into account when balancing sample distribution across labels_

Does this mean that `order` should be set to the maximum number of labels per data instance? Assuming a label distribution like the following, would 6 be the appropriate value for `order`? or 3 would be best as it is "simpler" while it encompass the vast majority of datapoints?

> Number of datapoints with 1 label(s): 3066
> Number of datapoints with 2 label(s): 761
> Number of datapoints with 3 label(s): 212
> Number of datapoints with 4 label(s): 49
> Number of datapoints with 5 label(s): 7
> Number of datapoints with 6 label(s): 1

**b)** In the `iterative_train_test_split` method there is no `order` parameter. Digging into the code, I found that `order=2` hardcoded. Let's assume I aim for a train-test split of a dataset with the label distribution listed above. If my assumption in a) is correct, wouldn't it make sense to allow to choose a train-test split with a desired `order` ?

**Comment by Jopepato** [comment]
Hi! Yes it should be included in the `iterative_train_test_split` the order parameter. I will make the changes. Thank you for seeing it. Also as you say in #160 it would be better to improve the documentation as well. Thanks!
Also I checked into the code and the `iterative_train_test_split` uses the `IterativeStratification` inside so it would accept the same type of data as `IterativeStratification`
